### PR TITLE
Add spider for Teresina/PI

### PR DESCRIPTION
**Pull request by alfakini**
Hey people,

In this PR we add the spider to collect data from [Teresina/PI](http://www.dom.teresina.pi.gov.br/lista_diario.php).

### CODE REVIEW

File: processing/data_collection/gazette/spiders/sc_joinville.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import dateparser
```
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/sc_joinville.py]: This file may not be here. It is not related to your PR that references only Teresina-PI spider.

File: processing/data_collection/gazette/spiders/pi_teresina.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            yield Gazette(
+                date=date,
+                file_urls=file_urls,
+                is_extra_edition=False,
+                municipality_id=self.MUNICIPALITY_ID,
+                power='executive',
+                scraped_at=datetime.utcnow(),
+            )
+
+        next_page_path = response.css(self.NEXT_PAGE_CSS).extract_first()
```
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/pi_teresina.py]: Use `urljoin` (https://doc.scrapy.org/en/latest/topics/request-response.html#scrapy.http.Response.urljoin) method from the response to handle relative links:
```
for next_page_url in response.css(self.NEXT_PAGE_CSS).extract():
    yield Request(response.urljoin(next_page_url))
```

File: processing/data_collection/gazette/spiders/pi_teresina.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    ATTACHMENT_CSS = 'td:last-child a::attr(href)'
+    DATE_CSS = 'td:nth-child(2)::text'
+    GAZETTE_ROW_CSS = '.table tbody tr'
+    NEXT_PAGE_CSS = '.box-paginacao a.paginacao:last-child::attr(href)'
+    PDF_HREF_CSS = 'td:nth-child(3) a::attr(href)'
+
+    allowed_domains = ['www.dom.teresina.pi.gov.br']
+    name = 'pi_teresina'
+    start_urls = [f'{GAZETTE_URL}/lista_diario.php']
```
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/pi_teresina.py]: As you are using `urljoin` to follow the pagination (see my comment below), I don't see the need for `GAZETTE_URL` variable. Just include the full URL in `start_urls` list.

Comment (alfakini) [reply processing/data_collection/gazette/spiders/pi_teresina.py]: Done @rennerocha! 👍

**Comment by giovanisleite** [comment]
Hey, @alfakini, remember to update the [cities.md](https://github.com/okfn-brasil/diario-oficial/blob/master/CITIES.md)

**Comment by alfakini** [comment]
@rennerocha @giovanisleite Done with the refactoring and suggestion 👍

**Comment by jvanz** [comment]
@alfakini mark the change requests as resolved. @rennerocha, does it look good to you?

**Comment by giovanisleite** [comment]
This PR can be closed (Cc. @jvanz)

**Pull request closed without merging**